cosmos/cosmos-sdk
# crypto/keyring: fix offline keys migration

## Description
closes: #8633 
I tested this branch against gaia, and works for migrating multisig keys. I'm not sure about the ledger though, having issues connecting mine. Would be nice if someone can test this.

## Steps for testing
```
1) cosmos-sdk: git checkout sahith/fix-key-multisig

2) gaia: git checkout v2.0.15 && make install

3) create local keys, multisig key and a ledger key

4) gaia: git checkout v4.0.3

5) add the following replace directive to go.mod in gaia - 
       replace github.com/cosmos/cosmos-sdk => <path to cosmos-sdk with specified branch>

6) make install

7) gaiad keys migrate --home <path to gaiacli(default: $HOME/.gaiacli)>
```
---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.
